File: js/2688.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n').map(Number);

// dp[i][j] : i번째 자릿수 값이 j일때 줄어들지 않는 수의 개수
const dp = Array(65)
  .fill()
  .map(() => Array(10).fill(0));

for (let j = 0; j < 10; j++) {
  dp[1][j] = 1;
}

for (let i = 2; i < 65; i++) {
  for (let j = 0; j < 10; j++) {
    for (let k = j; k < 10; k++) {
      dp[i][j] += dp[i - 1][k];
    }
  }
}

const results = [];

for (let i = 1; i < inputs.length; i++) {
  const n = inputs[i];
  const result = dp[n].reduce((a, b) => a + b);
  results.push(result);
}

console.log(results.join('\n'));

File: c++/2688.cpp
#include <iostream>

using namespace std;

typedef long long ll;

ll dp[65][10];  // dp[i][j] : i번째 자릿수의 값이 j일때 줄어들지 않는 수의 개수

int main() {
  freopen("example.txt", "r", stdin);
  int T;
  cin >> T;

  for (int j = 0; j < 10; j++) {
    dp[1][j] = 1;
  }

  for (int i = 2; i < 65; i++) {
    for (int j = 0; j < 10; j++) {
      for (int k = j; k < 10; k++) {
        dp[i][j] += dp[i - 1][k];
      }
    }
  }

  while (T--) {
    int n;
    cin >> n;

    ll result = 0;
    for (int j = 0; j < 10; j++) {
      result += dp[n][j];
    }
    cout << result << '\n';
  }

  return 0;
}